snippets/permutations.md

```diff
@@ -0,0 +1,19 @@
+### permutations
+
+Returns array of permutations of given array.
+
+[See more](https://en.wikipedia.org/wiki/Fixed-point_combinator) combinators and Lambda calculus.
+
+```js
+const permutate = arr => arr.reduce(
+  U(f => (acc, item, key, arr) =>
```

Comment:
I did not know if I could, I was willing to add a kind of chapter about combinators, and how to define functions around them, but maybe it could be an entire new repo